dataflow/flex-templates/pipeline_with_dependencies/README.md

```diff
@@ -0,0 +1,169 @@
+
+# Dataflow flex template: a pipeline with dependencies and a custom container image.
+
+This project is created to illustrate the following setup of a Dataflow Python pipeline:
+- The pipeline is a package that consists of [multiple files](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#multiple-file-dependencies).
+- The pipeline has at least one dependency that is not provided in the default Dataflow runtime environment.
+- The workflow uses a [custom container image](https://cloud.google.com/dataflow/docs/guides/using-custom-containers) to preinstall dependencies and to define the pipeline runtime environment.
+- The workflow uses a [Dataflow Flex Template](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates) to control the pipeline submission environment.
+- The runtime and submission environment use same set of Python dependencies and can be created in a reproducible manner.
+
+To illustrate this setup, we use a pipeline that does the following:
+
+1. Finds the longest word in an input file
+2. Creates a [FIGLet text banner](https://en.wikipedia.org/wiki/FIGlet) from of it using [pyfiglet](https://pypi.org/project/pyfiglet/)
+3. Outputs the text banner in another file
+
+
+## The structure of the example
+
+The pipeline package is comprised of the `my_package` directory and the `setup.py` file. The package defines the pipeline, the pipeline dependencies, and the input parameters. You can define multiple pipelines in the same package. The `my_package.launcher` module is used to submit the pipeline to a runner.
```

Comment:
Maybe just my preference: I feel like the directory structure is easier to follow (e.g., https://github.com/google/dataflow-ml-starter/tree/main?tab=readme-ov-file#directory-structure)

Reply:
Yes, I like your structure and it seems to me that this example follows almost same structure. the only notable difference I see is that you fold the entrypoint inside the launcher: https://github.com/google/dataflow-ml-starter/blob/2c591d3d274f1ef9f2f79db026093cbf2b7d36d5/my_project/run.py#L97

I also considered that and went back-and-forth a few times. I finally chose to keep entrypoint outside 
1) for consistency with beam docs and juliaset example: https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#multiple-file-dependencies (recently updated)
2) some users supply their pipeline in a package, but pass a package tarball via --extra_package and don't use --setup_file option. in such setup, source of the package doesn't need to in the launch environment, as long as the package is installed, however to launch the pipeline from flex template, one needs to provide a py file with the entrypoint: FLEX_TEMPLATE_PYTHON_PY_FILE=/path/to/launcher.py . If that top-level entrypoint lives separately from the package it might be easier to accomodate both setups.

Do you see any other significant differences?

Reply:
I noticed this but I think it is fine. The ML starter one tries to keep all the python code in one place. Your way is also fine. :)